jetty-websocket/websocket-javax-client/src/main/java/org/eclipse/jetty/websocket/javax/client/JavaxWebSocketClientContainerProvider.java

```diff
@@ -50,31 +51,43 @@ public static void stop(WebSocketContainer container) throws Exception
      * </p>
      */
     @Override
-    protected WebSocketContainer getContainer()
+    public WebSocketContainer getContainer()
     {
-        // See: https://github.com/javaee/websocket-spec/issues/212
-        // TODO: on multiple executions, do we warn?
-        // TODO: do we care?
-        // TODO: on multiple executions, do we share bufferPool/executors/etc?
-        // TODO: do we want to provide a non-standard way to configure to always return the same clientContainer based on a config somewhere? (system.property?)
-
         JavaxWebSocketClientContainer clientContainer = new JavaxWebSocketClientContainer();
+        registerShutdown(clientContainer);
+        return clientContainer;
+    }
```

Comment:
Don't duplicate this code... can you call `getContainer(null)` instead?

Reply:
The implementations are slightly different, `new JavaxWebSocketClientContainer()` will set the name of the `HttpClient` but `new JavaxWebSocketClientContainer(null)` won't do this.

But I don't think we should be doing this because it is done by the `HttpClientProvider`.